modules/renter/download.go

```diff
@@ -493,11 +518,26 @@ func (r *Renter) managedScheduleNewChunks(ds *downloadState) {
 			continue
 		}
 
+		// Determine whether the download should be sped up.
+		su := (nextChunk.download.numChunks - nextChunk.index) <= uint64(numFastChunks)
+		var np = nextChunk.download.erasureCode.MinPieces()
+		if su { // If speed up required.
+			min := np
+			max := nextChunk.download.erasureCode.NumPieces()
+			n := int(math.Ceil(float64(min) * float64(maxActivePPCMultiplier)))
+			if n > max { // Cap the number of pieces downloaded at the total number of pieces.
+				n = max
+			}
+
+			np = n
+		}
+
+		fmt.Printf("Pieces concurrently downloaded: %d\n", np)
 		// Add an incomplete chunk entry for every piece of the download.
-		for i := 0; i < nextChunk.download.erasureCode.MinPieces(); i++ {
+		for i := 0; i < np; i++ {
 			ds.incompleteChunks = append(ds.incompleteChunks, nextChunk)
 		}
-		ds.activePieces += nextChunk.download.erasureCode.MinPieces()
+		ds.activePieces += np
```

Comment:
I'm pretty amazed that this update was really this easy. Looks like there's at least one corner case remaining though.